sklearn/utils/metaestimators.py

```diff
@@ -7,10 +7,54 @@
 from functools import update_wrapper
 import numpy as np
 from ..utils import safe_indexing
+from ..externals import six
+from abc import ABCMeta
+from ..base import BaseEstimator
 
 __all__ = ['if_delegate_has_method']
 
 
+class _BaseComposition(six.with_metaclass(ABCMeta, BaseEstimator)):
```

Comment:
Do you mean I should also modify Pipeline and FeatureUnion to use _BaseComposition?